frontend/src/bundles/studio/components/video-menu/components/menu-body/menu-body.tsx

```diff
@@ -16,21 +16,36 @@ type Properties = {
     title: string | React.ReactNode;
     onClose: () => void;
     onChatOpen?: () => void;
+    chatModalReference: React.RefObject<HTMLDivElement>;
+    scriptModalReference: React.RefObject<HTMLDivElement>;
 };
 
 const MenuBody: React.FC<React.PropsWithChildren<Properties>> = ({
     title,
     children,
     onClose,
     onChatOpen,
+    chatModalReference,
+    scriptModalReference,
 }) => {
-    const menuReference = useRef<HTMLDivElement | null>(null);
+    const menuBodyReference = useRef<HTMLDivElement | null>(null);
 
     useEffect(() => {
         const handleClickOutside = (event: MouseEvent): void => {
+            const isInsideMenuBody = menuBodyReference.current?.contains(
```

Comment:
I think it makes sense to move it to video-menu, so the code is managed in one place.